src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java

```diff
@@ -135,6 +139,39 @@ public void onStartStageStep(@Nonnull StepStartNode stepStartNode, @Nonnull Stri
         }
     }
 
+    @Override
+    public void onStartCreateSpanStep(@Nonnull StepStartNode stepStartNode, @Nonnull WorkflowRun run) {
+        try (Scope ignored = setupContext(run, stepStartNode)) {
+            verifyNotNull(ignored, "%s - No span found for node %s", run, stepStartNode);
+
+            String createSpanName = getStepName(stepStartNode.getDescriptor(), "createSpan");
+            String stepType = getStepType(stepStartNode.getDescriptor(), "step");
+            JenkinsOpenTelemetryPluginConfiguration.StepPlugin stepPlugin = JenkinsOpenTelemetryPluginConfiguration.get().findStepPluginOrDefault(stepType, stepStartNode);
+
+            final Map<String, Object> arguments = ArgumentsAction.getFilteredArguments(stepStartNode);
+            final String spanAttribute = (String) arguments.getOrDefault("name", createSpanName);
+            final String spanName = Strings.isNullOrEmpty(spanAttribute) ? createSpanName : spanAttribute;
```

Comment:
I don't catch what is used as the Otel Span name if the script didn't provide a `name` invoking `createSpan()`. The unit test suggest that `"Create a Span"` would be used.

I'm wondering what's the standard way in Jenkins pipeline steps to deal with missing "required" parameters. 

I'm also wondering if we could say that a missing `name` would imply the logic to add the specified span àttributes`to children spans but I don't think it's a good idea, I feel introducing `setSpanAttributes(strategy: current_span|children_spans, attributes: ...)` would be more intuitive

Reply:
> I don't catch what is used as the Otel Span name if the script didn't provide a name invoking createSpan(). The unit test suggest that "Create a Span" would be used.

That's correct, `Create a Span` is defined in the StepDescriptor for the CreateSpan step. That's the approach we use by default, to use what the descriptor provides if possible.

> I'm wondering what's the standard way in Jenkins pipeline steps to deal with missing "required" parameters.

We can enforce to be mandatory, though I didn't see it was required that's the reason I let the possibility to be empty.

> I'm also wondering if we could say that a missing name would imply the logic to add the specified span àttributesto children spans but I don't think it's a good idea, I feel introducing setSpanAttributes(strategy: current_span|children_spans, attributes: ...)` would be more intuitive

I agree